.internal-ci/util/.shared_functions

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Copyright (c) 2018-2023 The MobileCoin Foundation
+
+shopt -s extglob
+
+# 1: semver tag
+# returns: network tier dev|test|main
+get_network_tier()
+{
+    case "${1}" in
+        v+([0-9])\.+([0-9])\.+([0-9])-test )
+            echo "Found Test version ${1}" >&2
+            echo "test"
+        ;;
+        v+([0-9])\.+([0-9])\.+([0-9]) )
+            echo "Found Main version ${1}" >&2
+            echo "main"
+        ;;
+        * )
+            echo "Found Development version ${1}" >&2
+            echo "dev"
+        ;;
+    esac
+}
+
+# 1: semver tag
+# returns: major version
+get_major_version()
+{
+    # Trim off version from - "dev" metadata
+    version=${1%%-*}
+    echo "DEBUG: version: $version" >&2
+    # Trim off the "patch"
+    v_major_minor=${version%.*}
+    echo "DEBUG: v_major_minor $v_major_minor" >&2
+    # Trim off the minor
+    v_major=${v_major_minor%.*}
+    echo "DEBUG: v_major $v_major" >&2
```

Comment:
:upside_down_face: is there a reason not to just stop at the first `.` the intermediate trimmings seem unecessary
```suggestion
    # Trim off version from - "dev" metadata
    v_major=${1%%.*}
```

Reply:
And now I remember why I was breaking it down from the back up.  This is valid semver and used in some places like the dev builds to add metadata but reduce char count.  `v0-v4-0-smoketest-release.1753.sha-33ea8c0f`

Splitting on the first `.` gives you `v0-v4-0-smoketest-release` which isn't the version.

Reply:
oh good to know maybe a comment in that function for future maintainers